### PR TITLE
Add /charts/series ext api

### PR DESCRIPTION
**Pull request by MikeShi42**

### CODE REVIEW

File: packages/api/src/routers/external-api/v1/charts.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    body: z.object({
+      series: z.array(externalQueryChartSeriesSchema).refine(
+        series => {
+          const groupByFields = series[0].groupBy;
+          return series.every(s => _.isEqual(s.groupBy, groupByFields));
+        },
+        {
+          message: 'All series must have the same groupBy fields',
+        },
+      ),
```
Comment (wrn14897) [inline packages/api/src/routers/external-api/v1/charts.ts:26]: oh this is cool

File: packages/api/src/routers/external-api/v1/charts.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      const teamId = req.user?.team;
+      const { endTime, granularity, startTime, seriesReturnType, series } =
+        req.body;
+
+      const internalSeries = series.map(s =>
+        translateExternalSeriesToInternalSeries({
+          type: 'time', // just to reuse the same fn
+          ...s,
+        }),
+      );
```
Comment (wrn14897) [inline packages/api/src/routers/external-api/v1/charts.ts:44]: nit: can do this after team validation

File: packages/api/src/utils/externalApi.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        where,
+        groupBy,
+        sortOrder,
+        content,
+        numberFormat,
+      } = s;
+
+      return {
+        type,
+        data_source: table === 'metrics' ? 'metrics' : 'events',
```
Comment (wrn14897) [inline packages/api/src/utils/externalApi.ts:159]: do we want to make this guy camelcase ?

Comment (MikeShi42) [reply packages/api/src/utils/externalApi.ts:159]: 🤦 yeah good point... there's a few places where I introduced the snake case. Let me do it in a follow up PR after this and the TF one is merged so I can change it everywhere at once.

**Review by wrn14897 — APPROVED** [review]
lgtm

**Pull request merged**